### ISSUE TITLE
Autocomplete does not work through type aliases

### ISSUE THREAD

**Issue by TedDriggs** [racer-rust/racer#641]
Static and instance methods are not completed for type aliases. Example below:

```rust
#[derive(Debug, Clone, Eq, PartialEq)]
/// An expression parameter
pub struct Param<T : Eq + Hash> {
    allowed: HashSet<T>
}

impl<T : Eq + Hash> Param<T> {
    pub fn new<V: Iterator<Item=U>, U: Into<T>>(allows: V) -> Param<T> {
        let mut ep = Param { allowed: HashSet::new() };
        for i in allows {
            ep.allowed.insert(i.into());
        }
        
        // return the constructed ExprParam
        ep
    }
    
    pub fn accepts<U>(&self, v: U) -> bool where T : PartialEq<U> {
        self.allowed.iter().any(|i| *i == v)
    }
}

// separate file in the same folder
pub type ExprParam = Param<ExprParamType>;
```

**Comment by runcom** [comment]
fixed by #666?

**Comment by TedDriggs** [comment]
Yes, it seems to be working as of 2.0.8.
